src/hw/hw_mngr.cpp

```diff
@@ -357,3 +407,166 @@ void HWManager::setFanSpeedDelayed()
         setFanSpeed();
     });
 }
+
+void HWManager::detectFansDelayed()
+{
+    static boost::asio::deadline_timer tmr(io);
+
+    //  if we has expired attempts, then stop fan detection and work in normal
+    //  state
+    if (++numErrorAttempts > maxErrorAttempts)
+    {
+        log<level::ERR>("FAN Detection",
+                        entry("WHAT=reached maxErrorAttempts in state=%d",
+                              int(detectFansState)));
+        detectFansState = FanState::normal;
+        return;
+    }
+
+    // this implicitly cancels the timer
+    tmr.expires_from_now(boost::posix_time::seconds(5));
+    tmr.async_wait([&](const boost::system::error_code& err) {
+        if (err == boost::asio::error::operation_aborted)
+        {
+            /* we were canceled*/
+            return;
+        }
+        else if (err)
+        {
+            log<level::ERR>("Timer error",
+                            entry("WHAT=%s", err.message().c_str()));
+            return;
+        }
+
+        detectFansByRpmSpeed();
+    });
+}
+
+void HWManager::detectFansByRpmSpeed()
+{
+    log<level::DEBUG>("FAN: process detection by maximum RPM",
+                      entry("SYSTEM=current state=%d", int(detectFansState)));
+
+    if (!processSystemFans())
+    {
+        detectFansState = FanState::init;
+        detectFansDelayed();
+        return;
+    }
+
+    switch (detectFansState)
+    {
+        case FanState::init:
+            log<level::INFO>("FAN: set all fans PWM to 100%");
```

Comment:
Didn't you intend to move it to `level::DEBUG`?

Reply:
Ok, no problem, I've moved all the information to the DEBUG level. But I'm sure this code could only have INFO and ERR messages, without DEBUG (IMHO)
Fixed.